fecgov/openFEC
# Update downloads to check for exact counts, refactor count logic

## Summary (required)

Resolves #4071

- Update downloads to check for exact counts
- Clean up tests
- Modify count logic to handle exact or estimated counts. Indicate whether counts are estimates for better testing and for future work: https://github.com/fecgov/openFEC/issues/4003
- Add test coverage for count logic

## How to test the changes locally

- Check out this branch
- Get downloads working locally: https://github.com/fecgov/openFEC/wiki/Set-up-Redis,--Celery-on-local-and-test-the-downloads
- Run your local CMS and point to your local API
- Test downloading all raw IEs: http://localhost:8000/data/independent-expenditures/?data_type=efiling&is_notice=true . You shouldn't get an error message like you do on production.
- Test other downloads
- Test estimated and exact counts (under and over 500K)

## Impacted areas of the application
List general components of the application that this PR will affect:

- Downloads
- Counts for all endpoints

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
Use actual counts for Schedule E efile endpoint | https://github.com/fecgov/openFEC/pull/4002
